### DISCUSSION TITLE
Access from the dockerd container to AWS ECR container image registry

### DISCUSSION THREAD

**Discussion by sysarch-repo** [lfn-cnti/testsuite#2031] [General]
How shall the testsuite be set up so that the dockerd process can communicate to AWS ECR. Currently the requests to AWS ECR are failing because of the missing login credentials. Example:

✖️  FAILED: Image size too large 🦖 ⚖👀
E, [2024-05-08 23:15:11 +00:00 #173348] ERROR -- cnf-testsuite-reasonable_image_size: invalid compressed_size: XXXXXXXXXXX.dkr.ecr.XXXXXXX.amazonaws.com/sysarch/dns:24.03.0 = '', Invalid Int64: ""

I am currently using the following workaround before triggering the AUT setup:

```
/home/ubuntu/.cnf-testsuite/cnf-testsuite install_dockerd

if [ $? -ne 0 ]; then
    echo "The installation of dockerd has failed."
    exit $?
fi

ECRPWD="$(aws ecr get-login-password --region $REGION)"
CMD="docker login $ACCOUNT.dkr.ecr.$REGION.amazonaws.com --username AWS --password $ECRPWD"
for i in $(seq 1 10); do
    kubectl exec -n cnf-testsuite dockerd -t -- $CMD 2> /dev/null
    
    if [ $? -ne 0 ]; then
        echo "The update of dockerd has failed. Re-trying..."
        sleep 15
    else
        echo "ECR credentials added for dockerd."
        break
    fi
done
```

With this, the container image size is passed as the dockerd process can communicate with the registry for the duration of the validity of the provided token.

I could even install the https://github.com/awslabs/amazon-ecr-credential-helper on the AWS worker nodes but also then I would need to set up the config.json in the dockerd container space to utilize the helper which would basically be just another workaround?

**Comment by taylor** [comment]
This can be an enhancement request in https://github.com/cnti-testcatalog/testsuite/issues
